### ISSUE TITLE
Offline Maps

### ISSUE THREAD

**Issue by beli3ver** [y20k/trackbook#25]
Hello,

It would be really nice, when this app could use offline maps, like c:geo or Osmand.
Thanks

Malte

**Comment by y20k** [comment]
Hi there,

Thanks for the suggestion!! 

Trackbook is supposed to be a very simple app. Just the most basic features. In part because I am a bit time constraint and even keeping the simple features working is time consuming.... and also in part because I really like simple apps.

I wrote that so that you don't think I am rude, when I am asking: **Are offline maps necessary?** Trackbook kind of works without any network connection. It records your tracks (at least it should). Okay. It does it on an empty map, but still you can record tracks without internet access and save them for later use.

I do not know all the ways people use the app. So what is you reason for this feature request: 

A) Is it the concern about unnecessary cell data usage 
B) or is it that you want to use the app outside of cell coverage and that you do want to look at the track while it is recording

Just curious.

Best regards

**Comment by beli3ver** [comment]
Hi there,

Thanks for your answer.

I wan to use the app in areas with no cell connection. So I use this app also to navigate through the area like Forrest or stuff like that. So I could download the offline map and then use it with this great app.

Malte

Am 27. August 2017 20:48:38 MESZ schrieb y20k <notifications@github.com>:
>Hi there,
>
>Thanks for the suggestion!! 
>
>Trackbook is supposed to be a very simple app. Just the most basic
>features. In part because I am a bit time constraint and even keeping
>the simple features working is time consuming.... and also in part
>because I really like simple apps.
>
>I wrote that so that you don't think I am rude, when I am asking: **Are
>offline maps necessary?** Trackbook kind of works without any network
>connection. It records your tracks (at least it should). Okay. It does
>it on an empty map, but still you can record tracks without internet
>access and save them for later use.
>
>I do not know all the ways people use the app. So what is you reason
>for this feature request: 
>
>A) Is it the concern about unnecessary cell data usage 
>B) or is it that you want to use the app outside of cell coverage and
>that you do want to look at the track while it is recording
>
>Just curious.
>
>Best regards
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/y20k/trackbook/issues/25#issuecomment-325216829

-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by y20k** [comment]
> I use this app also to navigate

Thanks for clearing that up. Interesting that you use Trackbook also for offroad "navigation". I am not sure I that is a good experience. I am only using it myself to record tracks. I think to make it not suck at navigation it would need more features like setting a goal marker, or even location search.

Anyway. I do not want to dismiss your idea right away. I need to think about it for a while.

**Comment by beli3ver** [comment]
Ok, I don't use it as offline navigation directly, but when I am on my way I check the map, so I must not start 1000 apps.
But it would be nice to load a gpx file and follow the track.

**Comment by y20k** [comment]
> But it would be nice to load a gpx file and follow the track.

That is probably not going to happen. I want to Trackbook to stay very simple, to be a single-purpose app. I should just be a movement recorder.

>> I need to think about it for a while.

Result -> There are two reasons why I am reluctant to extend the functionality of Trackbook:
* **Time:** Trackbook a hobby project. More complexity will bring more problems and will make the app harder to maintain. Given that I think it is better to work on the existing feature set and let other teams make the GPX Display app and the Offline Maps Navigation app.
* **I like single-purpose apps:** I understand that many people do not want to "start 1000 apps" (to quote you 😄). But I personally really like simple apps like calculators, todo lists, address books, webradio players. And I don't mind to open several apps to achieve several tasks, even if those tasks are related. I know that the following is not a fair comparison to your suggestions, so take it as a metaphor: I really dislike MS Outlook, although email, calendar, contact management etc. are related tasks.

I might still add offline maps functionality in the future though since that is a nice feature for a movement recorder after all. I would need to figure out an easy to use interface for that feature first. And I need to check how much time that would cost me.

**Comment by beli3ver** [comment]
So how could we help you 💃

**Comment by ort163** [comment]
I agree that this needs to be a *small* simple app. But would be nice if it can load a map "if there is file". Otherwise can be a blank. 
This app
https://gitlab.com/mvglasow/satstat
loads map from
http://download.mapsforge.org/

If possible...

**Comment by 532910** [comment]
SatStat is not a much larger.

**Comment by OmlineEditor** [comment]
almost 5 years have passed. you still haven't made the maps. I'm really looking forward to adding offline maps.

if you have a problem making offline maps, then make it possible to download all the map scales for the area that I'm looking at now. then I will go through all the villages near the city and all the maps will be loaded.
